### DISCUSSION TITLE
Migrating from resteasy to resteasy reacitve APIs give me a 404

### DISCUSSION THREAD

**Discussion by ramakaruturi** [quarkusio/quarkus#37949] [Community]
Hi There,

I have recently tried migrating a project of mine from Quarkus 2.x to 3.x and also switiching from resteasy to resteasy reactive.
While doing that i have observed that my APIs are no longer discoverable i.e. giving 404 after a successful build and deploy.

Upon some digging and trail and error i have narrowed it down and found out its the interfaces that were generated by OpenAPI generator in another jar that i was referring to implement in my project. I have also got a small reproducer here

Project #1
https://github.com/ramakaruturi/openapi-generated
This project uses OpenAPI generator to generate interfaces using JAX-RS and Jakarta EE as a jar file

Project #2
https://github.com/ramakaruturi/petstore-apis
This project consumes the jar file from #1 to implement the interfaces.

If you could please let me know what i could be possibly missing here that the APIs wouldnt be discovered ? 
The same setup used to work fine with Quarkus 2.7 and Resteasy.

TIA !!!!

**Comment by geoand** [comment]
cc @phillip-kruger

**Comment by phillip-kruger** [comment]
Cc @MikeEdgar

**Comment by MikeEdgar** [comment]
@ramakaruturi , just to clarify - is it that the API does not function at all after the upgrade or just that the OpenAPI documentation is not available?

**Reply by rama-karuturi** [reply]
The API doesn’t function at all…

**Reply by rama-karuturi** [reply]
Gives me a 404

**Reply by rama-karuturi** [reply]
Any luck yet were you able to run the reproducers ive shared ?

**Reply by MikeEdgar** [reply]
I haven't yet had a chance to run it. However, based on the description of the problem, I do think this is a RestEasy issue rather than an OpenAPI issue. 
cc: @geoand , @phillip-kruger

**Comment by gsmet** [comment]
Do you index the jar generated here with Jandex: https://github.com/ramakaruturi/openapi-generated ?

RESTEasy Reactive is heavily relying on Jandex to set up the endpoints so if your classes are not indexed, they won't be found by RESTEasy Reactive.
When having the endpoints in a separate jar, you need to make sure the jar is indexed as Quarkus only indexes automatically the main app classes.

See https://stackoverflow.com/questions/55513502/how-to-create-a-jandex-index-in-quarkus-for-classes-in-a-external-module for more information about that.
